pom.xml

```diff
@@ -179,9 +179,18 @@
                 </executions>
                 <configuration>
                     <java>
+                        <includes>
+                            <!-- include all java files in "java" folders under "src" -->
+                            <include>src/**/java/**/*.java</include>
+                        </includes>
+
+                        <excludes>
+                            <exclude>src/main/java/org/rumbledb/parser/*</exclude>
+                        </excludes>
                         <eclipse>
-                            <file>${basedir}/spotless-formatter-eclipse-jdt-configurations.xml</file>
+                            <file>${project.basedir}/spotless-formatter-eclipse-jdt-configurations.xml</file>
                         </eclipse>
+                        <removeUnusedImports/>
```

Comment:
This line handles the unused imports. If any unused import is found in the code and spotless:check is applied, it throws errors (I believe spotless:apply should remove unused imports but I haven't double checked)